modules/apps/portal-scheduler/portal-scheduler-test/src/testIntegration/java/com/liferay/portal/scheduler/internal/test/SchedulerEngineHelperTest.java

```diff
@@ -0,0 +1,411 @@
+						expectedLogs._logMessages, logEntry.getMessage()));
+
+				Throwable throwable = logEntry.getThrowable();
+
+				Assert.assertTrue(
+					ArrayUtil.contains(
+						expectedLogs._logMessages, throwable.getMessage()));
+			}
+		}
+	}
+
+	private void _refreshPortalSchedulerBundle() throws Exception {
+		for (Bundle bundle : _bundleContext.getBundles()) {
+			String symbolicName = bundle.getSymbolicName();
+
+			if (symbolicName.equals("com.liferay.portal.scheduler")) {
+				Bundle frameworkBundle = _bundleContext.getBundle(0);
+
+				FrameworkWiring frameworkWiring = frameworkBundle.adapt(
+					FrameworkWiring.class);
+
+				final DefaultNoticeableFuture<FrameworkEvent>
+					defaultNoticeableFuture = new DefaultNoticeableFuture<>();
+
+				frameworkWiring.refreshBundles(
+					Collections.<Bundle>singletonList(bundle),
+					new FrameworkListener() {
+
+						@Override
+						public void frameworkEvent(
+							FrameworkEvent frameworkEvent) {
+
+							defaultNoticeableFuture.set(frameworkEvent);
+						}
+
+					});
+
+				defaultNoticeableFuture.get();
+
+				return;
+			}
+		}
+
+		throw new IllegalStateException(
+			"Liferay Portal Scheduler is not deployed");
+	}
+
+	private AutoCloseable _registerCronJobConfiguration(String cronExpression) {
+		return _registerJobConfiguration(
+			new TestSchedulerCronJobConfiguration(cronExpression));
+	}
+
+	private AutoCloseable _registerIntervalJobConfiguration(
+		int interval, TimeUnit timeUnit) {
+
+		return _registerJobConfiguration(
+			new TestSchedulerIntervalJobConfiguration(interval, timeUnit));
+	}
+
+	private AutoCloseable _registerJobConfiguration(
+		SchedulerJobConfiguration schedulerJobConfiguration) {
+
+		Bundle bundle = FrameworkUtil.getBundle(
+			SchedulerEngineHelperTest.class);
+
+		BundleContext bundleContext = bundle.getBundleContext();
+
+		ServiceRegistration<?> serviceRegistration =
+			bundleContext.registerService(
+				SchedulerJobConfiguration.class, schedulerJobConfiguration,
+				null);
+
+		return serviceRegistration::unregister;
+	}
+
+	private AutoCloseable _swapSchedulerEngine() {
+		SchedulerEngine schedulerEngine =
+			ReflectionTestUtil.getAndSetFieldValue(
+				_schedulerEngineHelper, "_schedulerEngine",
+				(SchedulerEngine)ProxyUtil.newProxyInstance(
+					SchedulerEngine.class.getClassLoader(),
+					new Class<?>[] {SchedulerEngine.class},
+					(proxy, method, args) -> {
+						if (Objects.equals(method.getName(), "schedule")) {
+							_calledScheduleMethod = true;
+						}
+
+						return null;
+					}));
+
+		return () -> ReflectionTestUtil.setFieldValue(
+			_schedulerEngineHelper, "_schedulerEngine", schedulerEngine);
+	}
+
+	private static final String _LOG_CLASS =
+		"com.liferay.portal.scheduler.internal.SchedulerEngineHelperImpl";
+
+	private static BundleContext _bundleContext;
+
+	private boolean _calledScheduleMethod;
+
+	@Inject
+	private SchedulerEngineHelper _schedulerEngineHelper;
+
+	private static class ExpectedLogs {
```

Comment:
@marianoalvarosaiz All 4 usages of this inner class are using same `level` and `loggerClassName`. For now, no need to abstract it into an inner class. Please remove it, and directly pass the `String... expectedMessages` to `assertExpectedLogs()`.